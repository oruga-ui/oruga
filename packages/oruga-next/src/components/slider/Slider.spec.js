import { shallowMount } from '@vue/test-utils'
import OSlider from '@components/slider/Slider'

let wrapper

describe('OSlider', () => {
    beforeEach(() => {
        wrapper = shallowMount(OSlider)
    })

    it('is called', () => {
        expect(wrapper.exists()).toBeTruthy()
    })

    it('render correctly', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })

    it('manage props correctly', async () => {
        const newVariant = 'is-info'
        expect(wrapper.vm.newTooltipVariant).toBe(wrapper.vm.variant)
        wrapper.setProps({ tooltipVariant: newVariant })
        expect(wrapper.vm.newTooltipVariant).toBe(newVariant)

        const value1 = 3
        const value2 = 5
        const value = [value1, value2]
        const min = 1
        const max = 7
        wrapper.setProps({ value, min, max })
        await wrapper.vm.$nextTick()
        expect(wrapper.vm.value1).toBe(value1)
        expect(wrapper.vm.value2).toBe(value2)
        expect(wrapper.vm.minValue).toBe(value1)
        expect(wrapper.vm.maxValue).toBe(value2)

        expect(wrapper.vm.tickValues).toEqual([])
        wrapper.setProps({ ticks: true })
        await wrapper.vm.$nextTick()
        expect(wrapper.vm.tickValues).toEqual([2, 3, 4, 5, 6])

        expect(wrapper.vm.precision).toBe(0)
        wrapper.setProps({ step: 0.5 })
        await wrapper.vm.$nextTick()
        expect(wrapper.vm.precision).toBe(1)
    })
})
