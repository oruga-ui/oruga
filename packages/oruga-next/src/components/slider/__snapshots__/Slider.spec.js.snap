// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OSlider render correctly 1`] = `
<div class="o-slide">
    <div class="o-slide__track">
        <div class="o-slide__fill o-slide__fill--primary" style="width: 0%; left: 0%;"></div>
        <!---->
        <o-slider-thumb-stub value="0" variant="primary" tooltip="true" format="raw" role="slider" aria-valuenow="0" aria-valuemin="0" aria-valuemax="100" aria-orientation="horizontal"></o-slider-thumb-stub>
        <!---->
    </div>
</div>
`;
